transcendence-app/src/chat/chat.service.ts

```diff
@@ -180,4 +184,46 @@ export class ChatService {
     }
     return this.chatroomMemberService.addChatroomMember(chatroomId, userId);
   }
+
+  async updateChatroomMember(
+    authUser: User,
+    chatroomId: string,
+    userId: string,
+    updateChatroomMemberDto: UpdateChatroomMemberDto,
+  ): Promise<ChatroomMember | null> {
+    const authChatroomMember =
+      await this.chatroomMemberRepository.getByIdWithUser(
+        chatroomId,
+        authUser.id,
+      );
+    const chatroomMember = await this.chatroomMemberRepository.getByIdWithUser(
+      chatroomId,
+      userId,
+    );
+    if (!authChatroomMember || !chatroomMember) {
+      throw new NotFoundException();
+    }
+    if (chatroomMember.owner) {
+      throw new ForbiddenException();
+    }
+    if (authChatroomMember.owner) {
+      return this.chatroomMemberService.updateById(
+        chatroomId,
+        userId,
+        updateChatroomMemberDto,
+      );
+    }
+    if (
+      !authChatroomMember.admin ||
```

Comment:
I'm not very sure where are we going to designate the admins yet 😅 We don't have any screen defined where they are going to exist, right?

Reply:
We have a design proposal made by @Damrod, [here](https://www.figma.com/file/EFntd84Sl8NuHJQBrkqBVS/aollero-%2Fchat-v0.2?node-id=751%3A2184&t=Iirxid9OyqGNW8y0-0)
And a task for it, #207

Reply:
Maybe you are speaking about different things?
Chatroom admins vs site wide admins?
Maybe not?

Reply:
Perhaps. I was referring to chatroom admins :sweat_smile: